pkg/profile/git_repo.go

```diff
@@ -0,0 +1,46 @@
+package profile
+
+import (
+	"strings"
+
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GitRepoRequiresUpdate checks if the git repository resource requires updating
+func GitRepoRequiresUpdate(existingRes, newRes *sourcev1.GitRepository) bool {
+	switch {
+	case existingRes.Spec.URL != newRes.Spec.URL:
+		return true
+	case existingRes.Spec.Reference.Branch != newRes.Spec.Reference.Branch:
+		return true
+	default:
+		return false
+	}
+}
+
+func (p *Profile) makeGitRepository() (*sourcev1.GitRepository, error) {
```

Comment:
This never returns an `error`?

Reply:
this was already here, just moved it. Will update